error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1867,6 +1913,22 @@ Duration after(StepVerifier.LastStep step, Predicate<Throwable> predicate) {
     }
   }
 
+  /**
+   * Prefer {@link StepVerifier.LastStep#verifyErrorMatches(Predicate)} over more verbose
+   * alternatives.
+   */
+  static final class StepVerifierLastStepVerifyErrorMatchesAssertions {
+    @BeforeTemplate
+    void before(StepVerifier.LastStep step, Predicate<Throwable> predicate) {
+      step.expectError().verifyThenAssertThat().hasOperatorErrorMatching(predicate);
+    }
+
+    @AfterTemplate
+    void after(StepVerifier.LastStep step, Predicate<Throwable> predicate) {
+      step.verifyErrorMatches(predicate);
+    }
+  }
```

Comment:
This case is not exhaustive due to the overload of `StepVerifier.LastStep#expectedError`, but IMO covers sufficiently.